v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/BigQueryToElasticsearch.java

```diff
@@ -88,35 +124,82 @@ private static PipelineResult run(BigQueryToElasticsearchOptions options) {
+        pipeline
+            .apply(
+                "ReadFromBigQuery",
+                ReadBigQueryTableRows.newBuilder()
+                    .setOptions(options.as(BigQueryToElasticsearchOptions.class))
+                    .build())
 
-        /*
-         * Step #2: Convert table rows to JSON documents.
-         */
-        .apply("TableRowsToJsonDocument", ParDo.of(new TableRowToJsonFn()))
+            /*
+             * Step #2: Convert table rows to JSON documents.
+             */
+            .apply("TableRowsToJsonDocument", ParDo.of(new TableRowToJsonFn()));
 
-        /*
-         * Step #3: Apply UDF functions (if specified)
-         */
-        .apply(
-            TransformTextViaJavascript.newBuilder()
-                .setFileSystemPath(options.getJavascriptTextTransformGcsPath())
-                .setFunctionName(options.getJavascriptTextTransformFunctionName())
-                .build())
+    /*
+     * Step #3: Apply UDF functions (if specified)
+     */
+    PCollection<String> udfOut;
+    if (usePythonUdf) {
+      udfOut =
+          readJsonDocuments
+              .apply(
+                  "MapToRecord",
+                  PythonExternalTextTransformer.FailsafeRowPythonExternalUdf
+                      .stringMappingFunction())
+              .setRowSchema(PythonExternalTextTransformer.FailsafeRowPythonExternalUdf.ROW_SCHEMA)
+              .apply(
+                  "InvokeUDF",
+                  PythonExternalTextTransformer.FailsafePythonExternalUdf.newBuilder()
+                      .setFileSystemPath(options.getPythonExternalTextTransformGcsPath())
+                      .setFunctionName(options.getPythonExternalTextTransformFunctionName())
+                      .build())
+              .setRowSchema(
+                  PythonExternalTextTransformer.FailsafeRowPythonExternalUdf.FAILSAFE_SCHEMA)
+              .apply(
+                  MapElements.via(
+                      new SimpleFunction<>() {
+                        @Override
+                        public String apply(Row row) {
+                          String errorMessage = row.getValue("error_message");
+                          String stackTrace = row.getValue("stack_trace");
+                          if (!Strings.isNullOrEmpty(errorMessage)
+                              || !Strings.isNullOrEmpty(stackTrace)) {
+                            Object originalElement = row.getValue("original");
+                            throw new RuntimeException(
+                                String.format(
+                                    "Error applying UDF to the source record [%s], error message: [%s], stack trace: [%s]",
+                                    originalElement, errorMessage, stackTrace));
+                          }
+
+                          Row transformedRow = row.getValue("transformed");
+                          return transformedRow.getValue("message");
+                        }
```

Comment:
This should be a `RowToStringFn` class in `PythonExternalTextTransformer` similar to `RowToStringFailsafeElementFn`

Reply:
Addressed in https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1425